src/Http/Response.php

```diff
@@ -925,6 +925,12 @@ public function getStatusCode()
      */
     public function withStatus($code, $reasonPhrase = '')
     {
+        if ($code === null) {
+            return $this->_status;
```

Comment:
This read part and $code being NULL is invalid.

Reply:
You're right; statusCode used it to set the default status code to the response. Changed it to set the default response code to $new.